src/skai/model/data.py

```diff
@@ -567,7 +567,7 @@ class SkaiDatasetConfig(tfds.core.BuilderConfig):
 
 
 def decode_and_resize_image(
-    image_bytes: tf.Tensor | bytes, size: int) -> tf.Tensor:
+    image_bytes: Union[tf.Tensor, bytes], size: int) -> tf.Tensor:
```

Comment:
@jzxu 
Thank you for this comment.

## context
Originally, you provide container images to support python versions 3.7-3.11. 

One of the requirement for Dataflow jobs with python is that the local runtime versions of python and apache-beam sdk should match those in the container image (bullet points 3 and 5 - https://cloud.google.com/dataflow/docs/guides/using-custom-containers#troubleshooting). 

## Reason for change
At the moment, running the inference code from local python version < 3.10 breaks before sending the job because they do not allow this and the other type hints that's changed in this PR. Unless, we strictly want to provide support for python versions >= 3.10, any advice on that would be much appreciated.

Reply:
I see, thank you for the explanation. Are you aware of any hardship that requiring Python version >= 3.10 would have on Instadeep or any other organizations? If not, I'm inclined to just drop support for Python versions before 3.10. For example, I see that the default Python3 versions for Ubuntu and Debian Linux are both >= 3.10. The default Anaconda install is Python 3.11.

Reply:
We can drop support for versions earlier than 3.10 and be fine at least for inference on cpu/gpu.

Reply:
SGTM. We can also maybe convince some folks in cloud to build 3.10+ compatible TPU images for us.